components/workspacekit/cmd/rings.go

```diff
@@ -397,7 +397,7 @@ var ring1Cmd = &cobra.Command{
 
 		client, err := connectToInWorkspaceDaemonService(ctx)
 		if err != nil {
-			log.WithError(err).Error("cannot connect to daemon")
+			log.WithError(err).Error("cannot connect to daemon from ring1")
```

Comment:
Should it be ring2?

Reply:
I think it is correct because ring2 starts [here](https://github.com/gitpod-io/gitpod/pull/9951/files/d686941e005068b568897cb71413de82f399e3e2#diff-ef5b6a5c39ce2e00bf136797cd496a41f3c783ddc28811731b7d7b73b50e23a1R374) as another process from ring1. [This code](https://github.com/gitpod-io/gitpod/pull/9951/files/d686941e005068b568897cb71413de82f399e3e2#diff-ef5b6a5c39ce2e00bf136797cd496a41f3c783ddc28811731b7d7b73b50e23a1R374) calls itself (i.e., workspacekit) in a separate process. Double-fork is required before the container is created.

Reply:
Thanks for your kindly explanation @utam0k, attach the diagram for future reference
![image](https://user-images.githubusercontent.com/49380831/168001208-d05f0f95-7d99-4d35-9e25-fc31363c09bf.png)